sycl/include/CL/sycl/ONEAPI/reduction.hpp

```diff
@@ -508,30 +508,24 @@ class reduction_impl : private reduction_impl_base {
   }
 
   /// Constructs reduction_impl when the identity value is statically known.
-  // Note that aliasing constructor was used to initialize MRWAcc to avoid
```

Comment:
Why "aliasing constructor" was not enough?

Reply:
The idea of using aliasing constructor was to give the pointer to user's accessor to shared_ptr, but not release memory (i.e. user's accessor) when that shared_ptr is destroyed, at least that is how I understood it when wrote that code.

That was wrong approach for cases where user's accessor is destroyed before the accessor is needed in reduction computation. 
At this line: https://github.com/v-klochkov/llvm-test-suite/blob/public_vklochkov_reduction_ctor_acc_destroyed_early/SYCL/Reduction/reduction_nd_lambda.cpp#L41
MRWAcc shared_ptr holded a pointer to released memory.

Reply:
OK, thanks for explanation